src/cloudant/document.py

```diff
@@ -351,5 +401,5 @@ def put_attachment(self, attachment, content_type, data, headers=None):
             headers=headers
         )
         resp.raise_for_status()
-
+        self.fetch()
```

Comment:
I think it is the attachment metadata that we would need to manipulate in the local dictionary.  As in:
```json
"_attachments": {
    "blah.txt": {
      "content_type": "text/plain",
      "revpos": 11,
      "digest": "md5-bQ7PngG1dTfWpoCLVlLpPw==",
      "length": 1573,
      "stub": true
    }
  }
```

In the case of add/update attachments, I suppose we would be able to do that for all but the digest?  I'm not sure how the digest is populated.

In the case of delete we could remove the second `.fetch()` and programmatically remove the attachment entry from the local dictionary.  I will do that with `delete_attachment`.